src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -1007,31 +1007,32 @@ bool SQLStorage::loadCachedEcuManifest(const Uptane::EcuSerial& ecu_serial, std:
   return !empty;
 }
 
-void SQLStorage::storeMisconfiguredEcus(const std::vector<MisconfiguredEcu>& ecus) {
-  if (ecus.size() >= 1) {
-    SQLite3Guard db = dbConnection();
-
-    db.beginTransaction();
+void SQLStorage::saveMisconfiguredEcu(const MisconfiguredEcu& ecu) {
+  SQLite3Guard db = dbConnection();
 
-    if (db.exec("DELETE FROM misconfigured_ecus;", nullptr, nullptr) != SQLITE_OK) {
-      LOG_ERROR << "Can't clear misconfigured_ecus: " << db.errmsg();
-      return;
-    }
+  db.beginTransaction();
 
-    std::vector<MisconfiguredEcu>::const_iterator it;
-    for (it = ecus.begin(); it != ecus.end(); it++) {
-      auto statement = db.prepareStatement<std::string, std::string, int>(
-          "INSERT INTO misconfigured_ecus VALUES (?,?,?);", it->serial.ToString(), it->hardware_id.ToString(),
-          static_cast<int>(it->state));
+  auto statement = db.prepareStatement<std::string>("SELECT count(*) FROM misconfigured_ecus WHERE serial = ?;",
```

Comment:
We could get rid of the count and use `INSERT OR REPLACE` here. `serial` would be better off being a primary key instead of just "unique" but it should also work.

Reply:
> We could get rid of the count and use `INSERT OR REPLACE` here.

Good point.

> `serial` would be better off being a primary key instead of just "unique" but it should also work.

True, but then I have to add a migration. Is that worth it?

Reply:
I think it should work without so probably not.